oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/storage/model/StorageModels.java

```diff
@@ -113,7 +117,19 @@ private void retrieval(final Class<?> clazz,
 
         for (Field field : fields) {
             if (field.isAnnotationPresent(Column.class)) {
+                if (field.isAnnotationPresent(SQLDatabase.AdditionalEntity.OnlyAdditional.class)
+                    || field.isAnnotationPresent(SQLDatabase.AdditionalEntity.OriginAndAdditional.class)) {
+                    if (!record) {
+                        throw new IllegalStateException("Model [" + modelName + "] is not a Record, @SQLDatabase.AdditionalEntity only supports Record.");
+                    }
+                }
+
                 Column column = field.getAnnotation(Column.class);
+                if (field.isAnnotationPresent(SQLDatabase.AdditionalEntity.OnlyAdditional.class)
+                    && field.isAnnotationPresent(SQLDatabase.AdditionalEntity.OriginAndAdditional.class)) {
+                    throw new IllegalStateException("Model [" + modelName + "] column [" + column.columnName() +
+                                                        "]: @OnlyAdditional and @OriginAndAdditional cannot be used on the same column.");
+                }
```

Comment:
Please consider this https://github.com/apache/skywalking/pull/9098/files#r876753312 , having 2 annotations increases the complexity unnecessarily IMO, or please elaborate why we have to provide 2 annotations here

Reply:
considering

Reply:
Merged and reply on https://github.com/apache/skywalking/pull/9098#discussion_r876753312